applications/plugins/SofaCUDA/sofa/gpu/cuda/CudaStandardTetrahedralFEMForceField.cu

```diff
@@ -762,7 +762,7 @@ __device__ void StandardTetrahedralFEMForceFieldCuda_BoyceAndArruda_deriveSPKTen
         {
             result[j][k] = inversematrix[j][k] * (-1.0/3.0) * I1;
             result[j][k] = result[j][k] + ID[j][k];
-            result[j][k] = result[j][k] * (1.0/2.0) * pow(J, (real)(-2.0/3.0));
+            result[j][k] = result[j][k] * (1.0/2.0) * powf(J, (real)(-2.0/3.0));
```

Comment:
pow in float even if real is double?

Reply:
you are right that it seems that you need to use pow if double and powf if float.😢
Never tried as I never compile with double with CUDA. 🙄

Reply:
it was compiling fine with CUDA_DOUBLE enabled but I guess it was casting the double in float and then back to float.
I pushed a templated version which *should* use the double version if needed

Reply:
In fact I don't understand why std::pow was not working, it is supposed to handle double or float
https://en.cppreference.com/w/cpp/numeric/math/pow

Reply:
err... this is a CUDA file so you should look for:
 - https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html 
 - https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html

Reply:
👍